### DISCUSSION TITLE
How to check auth.role in a Function?

### DISCUSSION THREAD

**Discussion by KevTale** [supabase/supabase#3554] [Questions] [answered]
Hi,

I've created a function like so:

```
create function upvote (id_to_upvote text) 
returns void as $$
begin
  update painting_submissions
  set number_of_votes = number_of_votes + 1
  where cid = id_to_upvote;
end;
$$ language plpgsql security definer;
```

I want to add a check to see if `auth.role = 'authenticated'`.
I don't want to add an extra policy for the overall table, that's why I'm using a function instead.

Thanks!

**Comment by steve-chavez** [comment, marked as answer]
If you call the function through [rpc](https://supabase.io/docs/reference/javascript/rpc), you can just use `auth.role() = 'authenticated'` inside your function
